llama-index-integrations/vector_stores/llama-index-vector-stores-redis/llama_index/vector_stores/redis/base.py

```diff
@@ -18,122 +18,110 @@
 from llama_index.core.vector_stores.types import (
     BasePydanticVectorStore,
     MetadataFilters,
+    MetadataFilter,
     VectorStoreQuery,
     VectorStoreQueryResult,
 )
 from llama_index.core.vector_stores.utils import (
     metadata_dict_to_node,
     node_to_metadata_dict,
 )
-from llama_index.vector_stores.redis.utils import (
-    TokenEscaper,
-    array_to_buffer,
-    check_redis_modules_exist,
-    convert_bytes,
-    get_redis_query,
-)
+from llama_index.vector_stores.redis.schema import RedisVectorStoreSchema
+from llama_index.vector_stores.redis.utils import REDIS_LLAMA_FIELD_SPEC
 
-import redis
-from redis import DataError
-from redis.client import Redis as RedisType
-from redis.commands.search.field import VectorField
+from redis import Redis
 from redis.exceptions import RedisError
 from redis.exceptions import TimeoutError as RedisTimeoutError
 
-_logger = logging.getLogger(__name__)
+from redisvl.index import SearchIndex
+from redisvl.schema import IndexSchema
+from redisvl.query import VectorQuery, FilterQuery, CountQuery
+from redisvl.query.filter import Tag, FilterExpression
+from redisvl.schema.fields import BaseField
+from redisvl.redis.utils import array_to_buffer
+
+
+logger = logging.getLogger(__name__)
 
 
 class RedisVectorStore(BasePydanticVectorStore):
     stores_text = True
     stores_node = True
     flat_metadata = False
 
-    _tokenizer: Any = PrivateAttr()
-    _redis_client: Any = PrivateAttr()
-    _prefix: str = PrivateAttr()
-    _index_name: str = PrivateAttr()
-    _index_args: Dict[str, Any] = PrivateAttr()
-    _metadata_fields: List[str] = PrivateAttr()
     _overwrite: bool = PrivateAttr()
-    _vector_field: str = PrivateAttr()
-    _vector_key: str = PrivateAttr()
+    _return_fields: List[str] = PrivateAttr()
+    _vector_field_name: str = PrivateAttr()
+    _node_id_field_name: str = "id"
+    _doc_id_field_name: str = "doc_id"
+    _text_field_name: str = "text"
 
     def __init__(
         self,
-        index_name: str,
-        index_prefix: str = "llama_index",
-        prefix_ending: str = "/vector",
-        index_args: Optional[Dict[str, Any]] = None,
-        metadata_fields: Optional[List[str]] = None,
-        redis_url: str = "redis://localhost:6379",
+        schema: Optional[IndexSchema] = None,
+        redis_client: Optional[Redis] = None,
+        redis_url: Optional[str] = None,
         overwrite: bool = False,
         **kwargs: Any,
```

Comment:
one thing that might help smooth out usage for existing users -- if kwargs contains any of the old kwargs, raise an error pointing them to an updated notebook example or similar? 👀

Reply:
Great idea. I also have planned a `from_params` style init that could help here too. But I like your idea!

Reply:
For pointing to a notebook example, can I update that notebook in this same PR or separate after the package is released?

Reply:
Yes! Definitely update the notebook example in this PR :)